tests/Unit/CSSList/DocumentTest.php

```diff
@@ -142,4 +149,92 @@ public function insertContentBeforeAppendsIfSiblingNotFound(): void
         self::assertCount(4, $subject->getContents());
         self::assertSame([$bogusOne, $sibling, $bogusTwo, $item], $subject->getContents());
     }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksForNoContentsReturnsEmptyArray(): void
+    {
+        $subject = new Document();
+
+        self::assertSame([], $subject->getAllDeclarationBlocks());
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksCanReturnOneDirectDeclarationBlockContent(): void
+    {
+        $subject = new Document();
+
+        $declarationBlock = new DeclarationBlock();
+        $subject->setContents([$declarationBlock]);
+
+        $result = $subject->getAllDeclarationBlocks();
+
+        self::assertSame([$declarationBlock], $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksCanReturnMultipleDirectDeclarationBlockContents(): void
+    {
+        $subject = new Document();
+
+        $declarationBlock1 = new DeclarationBlock();
+        $declarationBlock2 = new DeclarationBlock();
+        $subject->setContents([$declarationBlock1, $declarationBlock2]);
+
+        $result = $subject->getAllDeclarationBlocks();
+
+        self::assertSame([$declarationBlock1, $declarationBlock2], $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksAlsoReturnsDeclarationBlocksWithinAtRuleBlockList(): void
```

Comment:
Niggly, but we're not *actually* writing an essay :)

The test method name should be able to stand alone.

```suggestion
    public function getAllDeclarationBlocksReturnsDeclarationBlocksWithinAtRuleBlockList(): void
```

Reply:
Unless I've missed something, this is not yet resolved.